tests/device_random_test.c

```diff
@@ -130,3 +130,47 @@ static int s_device_rand_buffer_distribution_fn(struct aws_allocator *allocator,
 }
 
 AWS_TEST_CASE(device_rand_buffer_distribution, s_device_rand_buffer_distribution_fn)
+
+static int s_device_rand_buffer_append_distribution_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)allocator;
+    (void)ctx;
+
+    /* Create array full of zeroes, but only partially fill it with randomness */
+    uint8_t array[DISTRIBUTION_PUT_COUNT + 100] = {0};
+    struct aws_byte_buf buf = aws_byte_buf_from_empty_array(array, sizeof(array));
+    ASSERT_SUCCESS(aws_device_random_buffer_append(&buf, DISTRIBUTION_PUT_COUNT));
```

Comment:
trivial: maybe we should add a test for n > UINT32_MAX

Reply:
nice, this unveiled some failure
not a BUG per-se, because we'd properly report the failure
but on my x64 Mac, the old code would fail if size exceeded INT32_MAX
probably other POSIX platforms would fail similarly
I changed it work work on the data in smaller chunks

Reply:
aaaaand I took the test out.
to push the boundaries it required 64bit machines to make a 4+GB allocation
that would make it impossible to run the unit tests on something like a raspberry pi 3